pkg/segment/segment.go

```diff
@@ -0,0 +1,153 @@
+package segment
+
+import (
+	"errors"
+	"fmt"
+	"github.com/openshift/odo/pkg/preference"
+	"github.com/openshift/odo/pkg/version"
+	"github.com/pborman/uuid"
+	"golang.org/x/term"
+	"gopkg.in/segmentio/analytics-go.v3"
+	"io/ioutil"
+	"net"
+	"os"
+	"os/user"
+	"path/filepath"
+	"strings"
+	"time"
+)
+
+// Writekey will be the API key used to send data to the correct source on Segment
+var WriteKey = "4xGV1HV7K2FtUWaoAozSBD7SNCBCJ65U"
+
+type Client struct {
+	segmentClient     analytics.Client
+	config            *preference.PreferenceInfo
+	telemetryFilePath string
+}
+
+func NewClient(config *preference.PreferenceInfo) (*Client, error) {
+	homeDir, _ := os.UserHomeDir()
+	return newCustomClient(config,
+		filepath.Join(homeDir, ".redhat", "anonymousId"),
```

Comment:
we already have `.odo` dir, canot we use that instead of `.redhat`? or is this something all other devtools projects do?

Reply:
You are correct, it is something that all the other devtools projects do. There are future plans to segregate telemetry data from multiple tools and it would help if the anonymousID is the same. And I think it also affects the cost of Segment. Segment charges based on the number of users tracked/month, so having this mechanism is cost effective.